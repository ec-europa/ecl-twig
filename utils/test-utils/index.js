const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
