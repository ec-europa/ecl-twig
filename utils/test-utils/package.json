{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
