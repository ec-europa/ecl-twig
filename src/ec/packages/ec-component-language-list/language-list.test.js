import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import { dataSplash, dataOverlay } from './demo/data';

describe('EC - Language List', () => {
  describe('Splash', () => {
    const template =
      '@ecl-twig/ec-component-language-list/ecl-language-list-splash.html.twig';
    const render = params => renderTwigFileAsNode(template, params);

    const options = {
      items: dataSplash.items,
      overlay: false,
      logo: {
        alt: dataSplash.logoAlt,
        path: '/static/logo--mute.svg',
      },
      icon_path: '/static/icons.svg',
    };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(options, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(options, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Overlay', () => {
    const template =
      '@ecl-twig/ec-component-language-list/ecl-language-list-overlay.html.twig';
    const render = params => renderTwigFileAsNode(template, params);

    const options = {
      items: dataOverlay.items,
      overlay: true,
      close_label: dataOverlay.closeLabel,
      title: dataOverlay.title,
      icon_path: '/static/icons.svg',
    };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(options, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(options, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
