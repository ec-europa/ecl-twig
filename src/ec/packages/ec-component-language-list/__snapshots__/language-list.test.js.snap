// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Language List Overlay renders correctly 1`] = `
<div
  class="ecl-language-list ecl-language-list--overlay"
>
  <div
    class="ecl-language-list__container ecl-container"
  >
    <div
      class="ecl-row"
    >
      <div
        class="ecl-language-list__close ecl-col-12 ecl-col-lg-8 ecl-offset-lg-2"
      >
        <button
          class="ecl-button ecl-button--ghost ecl-language-list__close-button"
          type="submit"
        >
          <span
            class="ecl-button__container"
          >
            <span
              class="ecl-button__label"
            >
              Close
            </span>
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--s ecl-button__icon ecl-button__icon--after"
              focusable="false"
            >
              <use
                xlink:href="/static/icons.svg#ui--close"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="ecl-language-list__title ecl-col-12 ecl-col-lg-8 ecl-offset-lg-2"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-language-list__title-icon"
          focusable="false"
        >
          <use
            xlink:href="/static/icons.svg#general--generic-lang"
          />
        </svg>
        Select your language
      </div>
    </div>
    <div
      class="ecl-row"
    >
      <div
        class="ecl-language-list__column ecl-col-12 ecl-col-lg-4 ecl-offset-lg-2"
      >
        <ul
          class="ecl-language-list__list"
        >
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_bg"
            >
              български
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_es"
            >
              español
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_cs"
            >
              čeština
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_da"
            >
              dansk
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_de"
            >
              Deutsch
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_et"
            >
              eesti
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_el"
            >
              ελληνικά
            </a>
          </li>
          <li
            class="ecl-language-list__item ecl-language-list__item--is-active"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-language-list__link"
              href="/example#language_en"
            >
              <span
                class="ecl-link__label"
              >
                English
              </span>
               
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/static/icons.svg#ui--check"
                />
              </svg>
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_fr"
            >
              français
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_ga"
            >
              Gaeilge
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_hr"
            >
              hrvatski
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_it"
            >
              italiano
            </a>
          </li>
        </ul>
      </div>
      <div
        class="ecl-language-list__column ecl-col-12 ecl-col-lg-4"
      >
        <ul
          class="ecl-language-list__list"
        >
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_lv"
            >
              latviešu
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_lt"
            >
              lietuvių
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_hu"
            >
              magyar
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_mt"
            >
              Malti
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_nl"
            >
              Nederlands
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_pl"
            >
              polski
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_pt"
            >
              português
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_ro"
            >
              română
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_sk"
            >
              slovenčina
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_sl"
            >
              slovenščina
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_fi"
            >
              suomi
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_sv"
            >
              svenska
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`EC - Language List Splash renders correctly 1`] = `
<div
  class="ecl-language-list ecl-language-list--splash"
>
  <header
    class="ecl-language-list__header"
  >
    <img
      alt="European Commission logo"
      class="ecl-language-list__logo"
      src="/static/logo--mute.svg"
    />
  </header>
  <div
    class="ecl-language-list__container ecl-container"
  >
    <div
      class="ecl-row"
    >
      <div
        class="ecl-language-list__column ecl-col-12 ecl-col-lg-4 ecl-offset-lg-2"
      >
        <ul
          class="ecl-language-list__list"
        >
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_bg"
            >
              български
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_es"
            >
              español
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_cs"
            >
              čeština
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_da"
            >
              dansk
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_de"
            >
              Deutsch
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_et"
            >
              eesti
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_el"
            >
              ελληνικά
            </a>
          </li>
          <li
            class="ecl-language-list__item ecl-language-list__item--is-active"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-language-list__link"
              href="/example#language_en"
            >
              <span
                class="ecl-link__label"
              >
                English
              </span>
               
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/static/icons.svg#ui--check"
                />
              </svg>
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_fr"
            >
              français
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_ga"
            >
              Gaeilge
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_hr"
            >
              hrvatski
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_it"
            >
              italiano
            </a>
          </li>
        </ul>
      </div>
      <div
        class="ecl-language-list__column ecl-col-12 ecl-col-lg-4"
      >
        <ul
          class="ecl-language-list__list"
        >
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_lv"
            >
              latviešu
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_lt"
            >
              lietuvių
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_hu"
            >
              magyar
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_mt"
            >
              Malti
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_nl"
            >
              Nederlands
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_pl"
            >
              polski
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_pt"
            >
              português
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_ro"
            >
              română
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_sk"
            >
              slovenčina
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_sl"
            >
              slovenščina
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_fi"
            >
              suomi
            </a>
          </li>
          <li
            class="ecl-language-list__item"
          >
            <a
              class="ecl-link ecl-link--standalone ecl-language-list__link"
              href="/example#language_sv"
            >
              svenska
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
